### PR TITLE
Adrian salazar/modules

### PR DESCRIPTION
**Pull request by AqSlz**
What does this PR do?

- This Pull Request would allow staff to see all the modules and problems of Edugator through a nice UI
- This UI allows users to add new modules if necessary and rename, or remove, existing modules
- The problems of each module are visible and you can edit them with just a click

The relevant files are currently under the `src/pages/modules` folder, in which `ModulesView.tsx` is where the base component lies. 

- `ModuleOptions.tsx`  is where the menu for renaming and deleting module actions
- `NewModule.tsx` is the file for the New Module dialog
- `types.ts` contains some interfaces for the Problem and Module constructs

So far, the information is not in Redux. I will be working on that on following PRs.

### CODE REVIEW

File: src/pages/modules/ModuleOptions.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
+import { IconButton, Menu, MenuItem } from '@material-ui/core';
+import { MoreVert } from '@material-ui/icons';
+
+const useStyles = makeStyles((theme: Theme) =>
```
[the remark this answers is not in the document]
Comment (AqSlz) [reply src/pages/modules/ModuleOptions.tsx:6]: For the first comment, `useStyles` what you see there is the way materialUI modifies the styles. The styles modified are pretty basic so far, only margin and padding mostly, so when we coordinate about a Theme for the whole website we can move this to a shared styles page so we can reuse them.

File: src/pages/modules/ModulesView.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
+import { Button, Typography, Accordion, AccordionDetails, AccordionSummary } from '@material-ui/core';
+import { ExpandMore, Edit, AssignmentTurnedIn } from '@material-ui/icons';
+import { IModule } from './types';
+import NewModuleDialog from "./NewModule";
+import ModuleOptions from "./ModuleOptions"
+import { LayoutContainer } from '../../shared/LayoutContainer';
+
+const useStyles = makeStyles((theme: Theme) =>
```
[the remark this answers is not in the document]
Comment (AqSlz) [reply src/pages/modules/ModulesView.tsx:10]: Same as above

File: src/pages/modules/ModulesView.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                </Accordion>
+              )
+
+            }) : <h1>Add some modules to get started. It doesn't show?</h1>
+        }
+      </div>
+    </LayoutContainer>
+
+  );
+}
```
[the remark this answers is not in the document]
Comment (AqSlz) [reply src/pages/modules/ModulesView.tsx:142]: Will add

File: src/pages/modules/NewModule.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { Button, TextField, DialogTitle, Dialog, Paper, Divider } from '@material-ui/core';
+import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles((theme: Theme) =>
```
[the remark this answers is not in the document]
Comment (AqSlz) [reply src/pages/modules/NewModule.tsx:5]: Same as above

File: src/pages/modules/NewModule.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            className={classes.button}
+            variant="outlined"
+          >
+            Add module
+          </Button>
+        </div>
+      </Paper>
+    </Dialog>
+  );
+}
```
[the remark this answers is not in the document]
Comment (AqSlz) [reply src/pages/modules/NewModule.tsx:74]: Will add

File: src/pages/modules/types.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    timeLimit: number,
+    memoryLimit: number,
+    buildCommand: string,
+}
+
+export interface IModule {
+    name: string,
+    number: number,
+    problemList: IProblem[],
+}
```
[the remark this answers is not in the document]
Comment (AqSlz) [reply src/pages/modules/types.ts:29]: Will add

File: src/pages/modules/ModulesView.tsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+import { Button, Typography, Accordion, AccordionDetails, AccordionSummary } from '@material-ui/core';
+import { ExpandMore, Edit, AssignmentTurnedIn } from '@material-ui/icons';
+import { IModule } from './types';
+import NewModuleDialog from "./NewModule";
+import ModuleOptions from "./ModuleOptions"
+import { LayoutContainer } from '../../shared/LayoutContainer';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
```
Comment (tanner-ropp) [inline src/pages/modules/ModulesView.tsx:12]: Is `root` being used in the document?

Comment (AqSlz) [reply src/pages/modules/ModulesView.tsx:12]: I will delete it

**Review by joshuadargan — DISMISSED** [review]
@Me on discord if you push any more changes. Lgtm

**Review by tanner-ropp — DISMISSED** [review]

**Review by tanner-ropp — APPROVED** [review]

File: src/pages/modules/NewModule.tsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    textfield: {
+      display: "block",
+      marginLeft: theme.spacing(3),
+      marginTop: theme.spacing(2),
+      margin: theme.spacing(1),
+      maxWidth: "70%",
+    },
+    dialog: {
```
Comment (joshuadargan) [inline src/pages/modules/NewModule.tsx:14]: Is this still needed?

**Review by joshuadargan — APPROVED** [review]
lgtm

**Pull request merged**